From d46b454758037d22e06bbac7d6968796479f63c3 Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Mon, 8 Mar 2021 11:20:11 +0530
Subject: [PATCH] Fixing stpncpy implementation.

sse2 version of stpncpy was returning incorrect
address.

Tracked-On: OAM-96397
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 libc/arch-x86_64/dynamic_function_dispatch.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libc/arch-x86_64/dynamic_function_dispatch.cpp b/libc/arch-x86_64/dynamic_function_dispatch.cpp
index 352635c36..eca295c5c 100644
--- a/libc/arch-x86_64/dynamic_function_dispatch.cpp
+++ b/libc/arch-x86_64/dynamic_function_dispatch.cpp
@@ -113,7 +113,7 @@ typedef char* stpncpy_func(char* __dst, const char* __src, size_t __n);
 DEFINE_IFUNC_FOR(stpncpy) {
     __builtin_cpu_init();
     if (__builtin_cpu_supports("avx2")) RETURN_FUNC(stpncpy_func, stpncpy_avx2);
-    RETURN_FUNC(strncpy_func, strncpy_generic);
+    RETURN_FUNC(stpncpy_func, stpncpy_generic);
 }
 
 typedef size_t strlen_func(const char* __s);
-- 
2.17.1

